c-layer/lib.cxx

```diff
@@ -162,6 +164,20 @@ void resolvable_unselect(const char* name, enum RESOLVABLE_KIND kind, struct Sta
   // it can look a bit strange to unset with higher causer then set, but USER is the highest priority that overwrite even locks
   // so we should be ok in Agama.
   selectable->unset(zypp::ResStatus::USER);
+  status->state = Status::STATE_SUCCEED;
+  status->error = NULL;
+}
+
+int run_solver(struct Status* status) noexcept {
+  try {
+    status->state = Status::STATE_SUCCEED;
+    status->error = NULL;
+    return zypp_ptr()->resolver()->resolvePool() ? 1 : 0;
```

Comment:
`resolvePool` returns `bool` "success"
on this line you return 1 to mean success but in the header and below you use the reversed "shell error" meaning

I take this as proof that `int` should be banned and C99 `bool` used instead :) in general
For this API, return `void` as nothing more than `Status` is needed

Reply:
well, Status is different from my POV...status handles errors ( so exceptions on libzypp layer ). And here return value if solver find solution or not..which is not error, but common functionality, so I do not agree with you that void should be used.
regarding that bool, if we switch to it, it should be consistent and we also need to check what bindgen generate, so I would let it for other PR.